deployability/modules/testing/README.MD

```diff
@@ -0,0 +1,341 @@
+
+      ```yaml
+      variables/agent-os
+      variables/manager-os
+      infra-provider
+      working-dir
+      tasks
+      ```
+
+      Pay attention to the tasks:
+
+      ```yaml
+      args
+      depends-on
+      ```
+
+      >Note: In args, configure the launcher's path correctly (main.py files in each module), and to fill `depends-on`, consider the steps of your test (allocation, provision, and test) 
+
+3. Execution of Command (local):
+
+  Execute the command by referencing the parameters required by the library (launcher).
+  
+  ```bash
+  python3 -m workflow_engine {.yaml fixture path} 
+  ```
+
+  Example
+
+  ```bash
+  python3 -m workflow_engine modules/workflow_engine/examples/dtt1-agents-poc.yaml
+  ```
+
+  > Note The command execution can also be mediated through Jenkins.
+
+
+If one wishes to execute the testing module without installing the workflow engine, they can proceed by using the launcher (module/testing/main.py):
+
+1. Execution
+
+  While in wazuh-qa/deployability
+
+  ```bash
+  python3 modules/testing/main.py --inventory '{{ inventory }}' --component {{ component }} --wazuh-version '{{ wazuh_version }}' --wazuh-revision '{{ wazuh_revision }}' --tests install --dependencies '{{ dependencies }}' --live {{ live }}
+
+  ```
+
+  Example:
+  ```bash
+  python3 modules/testing/main.py --inventory "/tmp/dtt1-poc/agent-linux-ubuntu-22.04-amd64/inventory.yaml" --component "agent" --wazuh-version "4.7.2" --wazuh-revision "1" --tests install --dependencies '{"manager":"/tmp/dtt1-poc/manager-linux-ubuntu-22.04-amd64/inventory.yaml"}' --live False
+  ```
+
+---
+
+### Technical documentation
+
+The testing module allows the execution of tests on agents and central components.
+
+Instructions can be received from the fixture and executed through the Workflow Engine or run through commands on an already provisioned infrastructure.
+
+In both cases, the following information will be required:
+
+```yaml
+# Generic agent test task
+- task: "run-agent-tests-{agent}"
+  description: "Run tests uninstall for the {agent} agent."
+  do:
+    this: process
+    with:
+      path: python3
+      args:
+        - modules/testing/main.py
+        - inventory: "{working-dir}/agent-{agent}/inventory.yaml"
+        - dependencies:
+          - manager: "{working-dir}/manager-{manager-os}/inventory.yaml"
+          - agent: "{working-dir}/agent-{agent}/inventory.yaml"
+        - tests: "install,basic_info,connection,register,repo,restart,stop,uninstall"
+        - component: "agent"
+        - wazuh-version: "4.7.2"
+        - wazuh-revision: "1"
+        - live: False
+```
+
+In the exposed fixture fragment, it can be observed that for the execution of the testing module launcher (`testing/main.py`), it is necessary to provide the inventory, dependencies, component, tests to execute, Wazuh version, Wazuh revision, and whether the repository is live or not (if not, it will look for information in `packages-dev` pre-release).
```

Comment:
> component -> components

Note: packages-dev should be between ``, removed to improve the reference to the changes

>  in `the` packages-dev pre-release `directory`).

Reply:
Done https://github.com/wazuh/wazuh-qa/pull/5005/commits/66983858545c260b04b271ca0cd647627fc80d08